### ISSUE TITLE
reformat readme.md file with index and description

### ISSUE THREAD

**Issue by themayurkumbhar** [piyushchaudhari04/DataStructure-Java#1]
Current readme.md file is empty.
should update to have index of all data structures with hyperlinks and descriptions.

@piyushchaudhari04 WDYT?

**Comment by piyushchaudhari04** [comment]
yes sure seems a good idea. Can you please add some content so that the same can be followed later
